src/test/kotlin/no/skatteetaten/aurora/boober/contracts/ContractBase.kt

```diff
@@ -0,0 +1,44 @@
+package no.skatteetaten.aurora.boober.contracts
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.jayway.jsonpath.DocumentContext
+import com.jayway.jsonpath.JsonPath
+import io.restassured.module.mockmvc.RestAssuredMockMvc
+import no.skatteetaten.aurora.boober.configureObjectMapper
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import java.io.File
+
+class ContractResponses(val jsonResponses: Map<String, DocumentContext>) {
+    inline fun <reified T : Any> response(responseName: String = jsonResponses.keys.first()): T {
+        val json = jsonResponses[responseName]?.jsonString()
+            ?: throw IllegalArgumentException("Invalid response name,  $responseName")
+        return configureObjectMapper(ObjectMapper()).readValue(json)
+    }
+}
+
+fun withContractResponses(baseTestObject: Any, fn: (responses: ContractResponses) -> Any) {
+    val responses = readJsonFiles(baseTestObject)
+    val controller = fn(responses)
+    setupMockMvc(controller)
+}
+
+private fun readJsonFiles(baseTestObject: Any): ContractResponses {
+    val baseName =
+        baseTestObject::class.simpleName
+            ?: throw IllegalArgumentException("Invalid base object, ${baseTestObject::class.simpleName}")
+    val folderName = "/contracts/${baseName.toLowerCase().removeSuffix("test")}/responses"
+    val content = baseTestObject::class.java.getResource(folderName)
+
+    val files = File(content.toURI()).walk().filter { it.name.endsWith(".json") }.toList()
+    val jsonResponses = files.associateBy({ it.name.removeSuffix(".json") }, { JsonPath.parse(it) })
+    return ContractResponses(jsonResponses)
+}
+
+private fun setupMockMvc(controller: Any) {
```

Comment:
Siden vi nå ikke bruker Spock, hadde det vært mulig å basere seg i større grad på `@WebMvcTest` (og ikke bruke RestAssuredMockMvc) slik at vi ikke trenger sette opp convertere og ObjectMappere selv? Vi hadde da trolig endt opp med å bruke `@Mock` i spring for å sette opp mocker, men jeg tror egentlig ikke det er vesentlig dårligere enn mockk.

Reply:
Ja, det er tanken. Jeg tenkte å ta det som neste steg, bare få alt over og få det til å funke med junit 5 og kotlin.